Sources/NIOSSH/Connection State Machine/SSHConnectionStateMachine.swift

```diff
@@ -152,6 +149,7 @@ struct SSHConnectionStateMachine {
             }
         case .keyExchange(var state):
             guard let message = try state.parser.nextPacket() else {
+                self.state = .keyExchange(state)
```

Comment:
ok, so far I caught it only in `.active` state and it breaks if we do not receive a full packet, only length, in parser we set state to `.waitingForBytes(length)` but due to CoW `SSHConnectionStateMachine` still has an old value of parser. And it's not safe to unwind the buffer as well, since `decryptFirstBlock` might be stateful as well. Now I need to figure out a test for this

Reply:
This was fixed in #102